rusqlite_migration/src/builder.rs

```diff
@@ -42,28 +42,25 @@ impl<'u> MigrationsBuilder<'u> {
     ///
     /// Panics if no migration with the `id` provided exists.
     #[must_use]
-    pub fn edit(mut self, id: usize, f: impl Fn(&mut M)) -> Self {
+    pub fn edit(mut self, id: usize, f: impl Fn(M) -> M) -> Self {
         if id < 1 {
             panic!("id cannot be equal to 0");
         }
-        f(self
-            .migrations
-            .get_mut(id - 1)
-            .expect("No migration with the given index"));
+        self.migrations[id - 1] = take(&mut self.migrations[id - 1]).map(f);
         self
     }
 
     /// Finalizes the builder and creates either a [`crate::Migrations`] or a
     /// [`crate::AsyncMigrations`] instance.
     pub fn finalize<T: FromIterator<M<'u>>>(self) -> T {
-        T::from_iter(self.migrations)
+        T::from_iter(self.migrations.iter().flatten().cloned())
```

Comment:
This `cloned` is really meh, but not sure how to get `M<'u>` out of `&M<'u>`. Maybe you have some idea @cljoly?

Reply:
I think I found a way to remove the `cloned`.

Reply:
Yeah, the `drain` idea seems to work.